docs/source/examples/classification_and_regression.rst

```diff
@@ -0,0 +1,324 @@
+
+Let's import all the needed packages.
+
+.. code-block:: python
+
+    import math
+    import os
+    import matplotlib.pyplot as plt
+    import numpy as np
+    import pandas as pd
+    import gdown
+    import zipfile
+    import cv2
+    from natsort import natsorted
+    import torch
+    import torch.nn as nn
+    import torch.nn.functional as F
+    import torch.optim as optim
+    from torch.autograd import Variable
+    import torchvision.datasets as datasets
+    import torchvision.models as models
+    import torchvision.transforms as tfms
+    from poutyne import set_seeds, Model, ModelCheckpoint, CSVLogger, Experiment, StepLR
+    from torch.utils.data import DataLoader, Subset, Dataset
+    from torchvision.utils import make_grid
+
+Training Constants
+==================
+
+.. code-block:: python
+
+    num_epochs = 15
+    learning_rate = 0.01
+    batch_size = 32
+    image_size = 224
+    valid_split_percent = 0.1
+    momentum = 0.5
+    set_seeds(42)
+    imagenet_mean = [0.485, 0.456, 0.406]  # mean of the ImageNet dataset for normalizing 
+    imagenet_std = [0.229, 0.224, 0.225]  # std of the ImageNet dataset for normalizing
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    print('The running processor is...', device) 
+
+CelebA Dataset
+==============
+
+We are going to use CelebA dataset for this experiment. The CelebA dataset  is a large-scale face attributes dataset which can be employed as the training and test sets for the following computer vision tasks: face attribute recognition, face detection, landmark (or facial part) localization, and face editing & synthesis.
+
+Fetching data 
+=============
+
+The section below consists of a few lines of codes that help us download celebA dataset from a public web source and unzip them.
+
+.. code-block:: python
+
+    data_root = 'data/celeba'
+    # Path to folder with the dataset
+    dataset_folder = f'{data_root}/img_align_celeba'
+
+    # URL for the CelebA dataset (aligned images, attributes, landmasrks)
+    url = 'https://graal.ift.ulaval.ca/public/celeba/img_align_celeba.zip'
+    attr_url = 'https://graal.ift.ulaval.ca/public/celeba/list_attr_celeba.txt'
+    land_mark_url = 'https://graal.ift.ulaval.ca/public/celeba/list_landmarks_align_celeba.txt'
+
+    # Path to download the dataset to
+    download_path = f'{data_root}/img_align_celeba.zip'
+    land_mark_path = f'{data_root}/list_landmarks_align_celeba.txt'
+    attr_path = f'{data_root}/list_attr_celeba.txt'
+
+    # Create required directories 
+    if not os.path.exists(data_root):
+    os.makedirs(data_root)
+    os.makedirs(dataset_folder)
+
+    # Download the dataset from the source
+    gdown.download(url, download_path, quiet=False)
+    gdown.download(land_mark_url, land_mark_path, quiet=False)
+    gdown.download(attr_url, attr_path, quiet=False)
```

Comment:
You don't need to use gdown since the links are not on Google Drive.

Reply:
i just found it easy for downloading. what do you recommend? 
how about "request.get" ?

Reply:
Yes, requests is good.